pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java

```diff
@@ -245,6 +250,16 @@ private synchronized void completeConnect(AuthData clientData) throws PulsarClie
         LOG.info("[{}] complete connection, init proxy handler. authenticated with {} role {}, hasProxyToBrokerUrl: {}",
             remoteAddress, authMethod, clientAuthRole, hasProxyToBrokerUrl);
         if (hasProxyToBrokerUrl) {
+            // Optimize proxy connection to fail-fast if the target broker isn't active
+            // Pulsar client will retry connecting after a back off timeout
+            if (service.getConfiguration().isCheckActiveBrokers()
+                    && !isBrokerActive(proxyToBrokerUrl)) {
+                ctx()
+                        .writeAndFlush(
+                                Commands.newError(-1, ServerError.ServiceNotReady, "Target broker isn't available."))
```

Comment:
what about reporting the address of the broker here ?

Reply:
It doesn't seem to be necessary to report to the client since the client has the information about the target broker.
I guess there could be logging with WARN level in this case?

Reply:
+1 for logging.
Actually it is better to not report the address to the client.
thanks

Reply:
I added the logging. PTAL